lib/api/src/grpc/conversions.rs

```diff
@@ -1754,6 +1755,36 @@ impl From<HnswConfigDiff> for segment::types::HnswConfig {
     }
 }
 
+impl From<StrictModeConfigDiff> for segment::types::StrictModeConfig {
+    fn from(value: StrictModeConfigDiff) -> Self {
+        Self {
+            enabled: value.enabled,
+            max_query_limit: value.max_query_limit.map(|i| i as usize),
+            max_timeout: value.max_timeout.map(|i| i as usize),
+            unindexed_filtering_retrieve: value.unindexed_filtering_retrieve,
+            unindexed_filtering_update: value.unindexed_filtering_update,
+            search_max_hnsw_ef: value.search_max_hnsw_ef.map(|i| i as usize),
+            search_allow_exact: value.search_allow_exact,
+            search_max_oversampling: value.search_max_oversampling.map(f64::from),
+        }
+    }
+}
```

Comment:
If we can construct a regular config from a diff, I don't see much reason to also have a diff type.